File: app/src/main/java/com/eccentricyan/pohe/presentation/base/BaseFragment.java
package com.eccentricyan.pohe.presentation.base;

import android.os.Bundle;
import android.support.annotation.CallSuper;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.eccentricyan.pohe.di.component.ActivityComponent;

import org.greenrobot.eventbus.EventBus;

import javax.inject.Inject;

import io.realm.Realm;

/**
 * Created by shiyanhui on 2017/04/15.
 */

public class BaseFragment extends Fragment {
    protected ActivityComponent component;

    @Inject
    protected EventBus eventBus;

    @Inject
    protected Realm realm;

    @CallSuper
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        this.component = ((BaseActivity) getActivity()).getComponent();
        this.component.inject(this);
    }

}

File: app/src/main/java/com/eccentricyan/pohe/presentation/base/BaseFragment.kt
package com.eccentricyan.pohe.presentation.base

import android.os.Bundle
import android.support.annotation.CallSuper
import android.support.v4.app.Fragment

import com.eccentricyan.pohe.di.component.ActivityComponent

import org.greenrobot.eventbus.EventBus

import javax.inject.Inject

import io.realm.Realm

/**
 * Created by shiyanhui on 2017/04/15.
 */

open class BaseFragment : Fragment() {
    protected var component: ActivityComponent? = null

    @JvmField
    @Inject
    var eventBus: EventBus? = null

    @JvmField
    @Inject
    var realm: Realm? = null

    @CallSuper
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        this.component = (activity as BaseActivity).component
        this.component!!.inject(this)
    }

}